esp-bootloader-esp-idf/src/partitions.rs

```diff
@@ -301,6 +301,48 @@ impl<'a> PartitionTable<'a> {
         }
         Ok(None)
     }
+
+    #[cfg(not(feature = "std"))]
+    /// Get the currently booted partition.
+    pub fn booted_partition(&self) -> Result<Option<PartitionEntry<'a>>, Error> {
+        // Read entry 0 from MMU to know which partition is mapped
+        //
+        // See <https://github.com/espressif/esp-idf/blob/758939caecb16e5542b3adfba0bc85025517db45/components/hal/mmu_hal.c#L124>
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "esp32")] {
+                let paddr = unsafe {
+                    ((0x3FF10000 as *const u32).read_volatile() & 0xff) << 16
+                };
+            } else if #[cfg(feature = "esp32s2")] {
+                let paddr = unsafe {
+                    (((0x61801000 + 128 * 4) as *const u32).read_volatile() & 0xff) << 16
+                };
+            } else if #[cfg(feature = "esp32s3")] {
+                // Revisit this once we support XiP from PSRAM for ESP32-S3
+                let paddr = unsafe {
+                    ((0x600C5000 as *const u32).read_volatile() & 0xff) << 16
+                };
+            } else if #[cfg(any(feature = "esp32c2", feature = "esp32c3"))] {
+                let paddr = unsafe {
+                    ((0x600c5000 as *const u32).read_volatile() & 0xff) << 16
+                };
+            } else if #[cfg(any(feature = "esp32c6", feature = "esp32h2"))] {
+                let paddr = unsafe {
+                    ((0x60002000 + 0x380) as *mut u32).write_volatile(0);
+                    (((0x60002000 + 0x37c) as *const u32).read_volatile() & 0xff) << 16
```

Comment:
Couldn't we use the PAC for all these?

Reply:
We could add those registers ofc - I just assumed it's not worth just for this sole use but we should assess that decision once we have more uses for that - I can create an issue for that if you like

Reply:
It would be great (https://github.com/esp-rs/esp-hal/issues/2748)